be/src/exec/sorting/merge_path.h

```diff
@@ -159,23 +164,29 @@ class Node {
     virtual void process_input(int32_t parallel_idx) = 0;
     virtual void process_input_done(){};
     virtual bool is_leaf() { return false; }
-    // Means the previous node reaches eos, no more data will flow to current node,
-    // but there still may be some data in the input buffer to be consumed.
-    virtual bool dependency_finished() = 0;
-    // Means based on dependency_finished state, all the data in the input buffer has been consumed.
-    virtual bool input_finished() = 0;
-    // Return true if current node is active but cannot provide more data at this moment.
-    virtual bool is_pending() { return false; }
+
```

Comment:
Another implementation is similar to spilling @stdpain,  merge sort operator hands over its work to merge path pipeline which scheduled on pipeline executors.  merge path pipelines looks like: 
level 0: MergePathOperator->Sink,
level 1: MergePathOperator->Sink,
...
level N: MergePathOperator->Sink,

The old MergeSortOperator's push_chunk method push data into  level 0's pipeline, its pull_chunk method pull data from the last level's pipeline.

Reply:
What's the good to use multiply pipelines to represents the merge tree?